src/components/Conversion/ConversionNOB.vue

```diff
@@ -0,0 +1,145 @@
+        <div v-if="onEditMode">
+          <p class="ma-0">
+            Provide a brief description of the nature of business (e.g., corner grocery store,
+            automotive repair service, landscaping, etc.).
+          </p>
+          <v-text-field
+            filled
+            persistent-hint
+            class="mt-5"
+            autocomplete="chrome-off"
+            label="Enter Nature of Business"
+            v-model="naicsText"
+            counter="300"
+            :rules="naicsRules"
+            @change="onChangeNaics($event)"
+          >
+          </v-text-field>
+          <div id="nob-confirm-container" class="mb-2">
+            <v-btn large color="primary" id="nob-done-btn" class="mr-2"
+              @click="onSubmitClicked()"
+            >
+              <span>Done</span>
+            </v-btn>
+            <v-btn large outlined color="primary" id="nob-cancel-btn"
+              @click="onCancelClicked()"
+            >
+              <span>Cancel</span>
+            </v-btn>
+          </div>
+        </div>
+
+        <div v-if="!onEditMode" class="summary-block d-flex justify-space-between align-center">
+          <span id="naics-summary">{{ naicsSummary }}</span>
+          <v-btn text color="primary" id="nob-change-btn" @click="onChangeClicked()">
+            <v-icon small>mdi-pencil</v-icon>
+            <span>Change</span>
+          </v-btn>
+        </div>
+      </v-col>
+    </v-row>
+  </section>
+</template>
+
+<script lang="ts">
+import { Action, Getter } from 'vuex-class'
+import { Component, Vue } from 'vue-property-decorator'
+import { ActionBindingIF } from '@/interfaces/'
+import { NaicsIF } from '@bcrs-shared-components/interfaces/'
+import { isEqual } from 'lodash'
+
+@Component({})
+export default class NatureOfBusiness extends Vue {
+  @Getter getCurrentNaics!: NaicsIF
+  @Getter hasNatureOfBusinessChanged!: boolean
+
+  @Action setNaics!: ActionBindingIF
+
+  // local variables
+  private onEditMode = false
+  private naicsRules = []
+  private naicsText = ''
+
+  /** Show naics value, description or (Not Entered) upon first render */
+  get naicsSummary (): string {
+    const code = this.getCurrentNaics.naicsCode
+    const desc = this.getCurrentNaics.naicsDescription
+    let summary = '(Not Entered)'
+    if (code && desc) {
+      this.naicsText = this.hasNatureOfBusinessChanged ? this.naicsText : `${code} - ${desc}`
+      summary = `${code} - ${desc}`
+    } else if (desc) {
+      this.naicsText = desc
+      summary = desc
+    }
+    return summary
+  }
+
+  /** The naics data on record for the business. */
+  get originalNaics (): NaicsIF {
+    return {
+      naicsCode: this.getCurrentNaics.naicsCode,
+      naicsDescription: this.getCurrentNaics.naicsDescription
+    }
+  }
+
+  /** Monitor naics text change */
+  private onChangeNaics (naicsText:string): void {
+    this.naicsRules = naicsText.length > 300 ? ['Maximum 300 characters reached'] : []
+  }
```

Comment:
First, I don't think you need to do this. I believe the v-text-field can handle this "max length" internally.

Second, the rules can be static. They don't need to be set here.

Reply:
I modified ConversionNOB.vue file in my latest commit.

Reply:
Looks good!